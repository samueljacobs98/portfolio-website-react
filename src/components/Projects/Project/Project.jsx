import React from "react";
import Links from "../Links/Links";
import img2048 from "../../../assets/images/2048screenshot.png";

import "./Project.scss";

const Project = () => {
  const skillJSX = (
    <>
      <p className="project__skill">HTML</p>
      <p className="project__skill">CSS</p>
      <p className="project__skill">JavaScript</p>
      <p className="project__skill">Figma</p>
    </>
  );

  return (
    <section className="project">
      <h3 className="project__heading">2048 Game</h3>
      <img className="project__img" src={img2048} alt="" />
      <p className="project__text">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc quisque
        orci imperdiet sit bibendum. Faucibus interdum adipiscing placerat diam
        velit ac commodo facilisi. Mauris eget tempor nibh phasellus. In
        scelerisque blandit non duis facilisi cursus et. Dictum amet integer non
        amet imperdiet dui, id varius sed. Risus risus senectus eu viverra nisl
        in velit, integer.
      </p>
      <div className="project__skills">{skillJSX}</div>
      <Links />
    </section>
  );
};

export default Project;
