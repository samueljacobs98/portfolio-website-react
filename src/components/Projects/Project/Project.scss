@use "../../../assets/scss/variables" as *;

.project {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  max-width: $project-width;
  max-height: 600px;
  &__img {
    max-width: $project-width;
    width: 100%;
    margin-bottom: $margin-project;
  }
  &__text {
    border: 2px solid $color-white;
    padding: $padding-project;
    overflow: auto;
    background-color: $colour-background;
  }
  &__para {
    font-family: $font-secondary;
  }
  &__skills {
    display: flex;
    justify-content: space-between;
    width: 100%;
    margin: $margin-project 0;
  }
}

@media screen and (max-width: 1060px) {
  .project {
    padding: $padding-project;
    order: 2;
  }
}
