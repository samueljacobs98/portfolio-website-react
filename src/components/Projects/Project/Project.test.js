import { render } from "@testing-library/react";
import Project from "./Project";

test("should match snapshot", () => {
  const data = {
    skills: ["", "", ""],
    title: "",
    desc: ["", ""],
    img: "",
    links: { github: "", deployed: "" },
  };
  const { container } = render(<Project data={data} />);
  expect(container).toMatchSnapshot();
});
