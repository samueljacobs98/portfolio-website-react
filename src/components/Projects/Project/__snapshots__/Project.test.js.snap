// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <section
    class="project"
  >
    <h3
      class="project__heading"
    />
    <img
      alt=""
      class="project__img"
      src=""
    />
    <div
      class="project__text"
    >
      <p
        class="project__para"
      />
      <br />
      <p
        class="project__para"
      />
    </div>
    <div
      class="project__skills"
    >
      <p
        class="project__skill"
      />
      <p
        class="project__skill"
      />
      <p
        class="project__skill"
      />
    </div>
    <div
      class="project-links"
    >
      <a
        class="project-links__container"
        href=""
      >
        <p
          class="project-links__text"
        >
          Code:
        </p>
        <img
          alt=""
          class="project-links__link"
          src="github.svg"
        />
      </a>
      <a
        class="project-links__container"
        href=""
      >
        <p
          class="project-links__text"
        >
          Preview:
        </p>
        <img
          alt=""
          class="project-links__link"
          src="preview.svg"
        />
      </a>
    </div>
  </section>
</div>
`;
